linker.ld

```diff
@@ -25,13 +25,17 @@ SECTIONS {
     . = 0x7c00;
     _stack_end = .;
 
-    .bootloader :
+    .bootloader_first_stage :
     {
         /* first stage */
         *(.boot-first-stage)
+    }
 
+    .bootloader :
+    {
         /* rest of bootloader */
         _rest_of_bootloader_start_addr = .;
+        *(.smp_trampoline)
```

Comment:
Quick question. How is it guaranteed that this will be in the first MB? If the rest of bootloader is large, wouldn't you dynamically search for a location and use that to store all the kernel rather than placing it in the first MB?

Reply:
We put that section together with the bootloader sections. The bootloader always lives in the first MB because otherwise it would not be able to load itself in real mode. So everthing uintl `__bootloader_end` needs to be in the first MB.